int32 number_of_objects
float32[] points_data
